src/flood_model/bg_flood_model.py

```diff
@@ -398,16 +398,13 @@ def run_bg_flood_model(
     # Get the valid BG-Flood Model directory
     bg_flood_dir = get_valid_bg_flood_dir()
     # Get the file path of the Hydro DEM for the catchment area
-    hydro_dem_path = dem_metadata_in_db.get_catchment_hydro_dem_filepath(engine, catchment_area)
-    # Use the resolution of the Hydro DEM if it is not provided
-    if resolution is None:
-        _, resolution = dem_metadata_in_db.get_hydro_dem_data_and_resolution(engine, catchment_area)
+    hydro_dem_path, _, _, resolution = get_dem_by_geometry(engine, catchment_area)
 
     # Prepare inputs for the BG-Flood Model
     prepare_bg_flood_model_inputs(
         bg_flood_dir=bg_flood_dir,
         model_output_path=model_output_path,
-        hydro_dem_path=hydro_dem_path,
+        hydro_dem_path=pathlib.Path(hydro_dem_path),
```

Comment:
```suggestion
        hydro_dem_path=hydro_dem_path,
```

I believe it's beneficial to enhance the `get_dem_by_geometry()` function by incorporating clearly defined type hints for its output. This way, in scenarios like when the `hydro_dem_path` is utilized within the `prepare_bg_flood_model_inputs()` function, there will be no necessity to employ `pathlib.Path()`.

Reply:
done